apps/meteor/app/api/server/v1/misc.ts

```diff
@@ -257,7 +258,7 @@ API.v1.addRoute(
 					const user = this.getUserFromParams();
 
 					// Respect the server's choice for using their real names or not
-					if (user.name && settings.get('UI_Use_Real_Name')) {
+					if (user.name && getMessagesLayoutPreference() !== 'username') {
```

Comment:
Can this check be moved into a helper function? I see that it's being checked more than 15 times the same thing, and I believe it would be better to move this into a separate function to reduce future maintenance.

`export const useRealName = (uid?: string) => getMessagesLayoutPreference(uid) !== 'username'`

Reply:
Fixed!